src/Microsoft.CodeAnalysis.Razor.Workspaces/WorkspaceMonitor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class WorkspaceMonitor
+    {
+        private readonly Workspace _workspace;
+
+        public WorkspaceMonitor(Workspace workspace)
+        {
+            if (workspace == null)
+            {
+                throw new ArgumentNullException(nameof(workspace));
+            }
+
+            _workspace = workspace;
+
+            _workspace.WorkspaceChanged += OnWorkspaceChanged;
+        }
+
+        public event Action TagHelperFileChanged;
+
+        public event Action ReferencesChanged;
```

Comment:
These events will probably need an `EventArgs`. The shape of these events implies that you want to know what file has changed, or what reference change as well.

I'd really suggest just consolidating these into one event, and have an `EventArgs` that describes the change. Even if that `EventArgs` class is empty now, we can add members to it in the future.